### PR TITLE
[C++] Competitive (LeetCode - Coin Change 2)

### PR DESCRIPTION
**Pull request by bhatipradeep**

### CODE REVIEW

**Comment by bhatipradeep** [comment]
@amanmehara, please review

File: C++/competitive/LeetCode/coin_change_2.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                
+                //add ways of forming an amount in both the above cases
+                dp[i][j]=a+b;
+                
+            }
+           
+        }
+        
+        //dp[coins.size][amount] represents the total number of ways for that required amount considering all denominations
+        return dp[coins.size()][amount];
```
Comment (amanmehara) [inline C++/competitive/LeetCode/coin_change_2.cpp:50]:
```suggestion
        int dp[5001] = {0};
        dp[0] = 1;
        for (const auto& coin: coins) {
            for (auto i = coin; i <= amount; i++) {
                dp[i] += dp[i - coin];
            }
        }
        return dp[amount];        
```

**Review by amanmehara — CHANGES_REQUESTED** [review]
I've suggested a smaller code which is more efficient.

File: C++/competitive/LeetCode/coin_change_2.cpp (hunk 1 above)
Comment (bhatipradeep) [reply C++/competitive/LeetCode/coin_change_2.cpp:50]: should I change my solution as per your recommended solution ?

Comment (bhatipradeep) [reply C++/competitive/LeetCode/coin_change_2.cpp:50]: Your solution is really optimised. Great!

Comment (amanmehara) [reply C++/competitive/LeetCode/coin_change_2.cpp:50]: sorry did not see the message.
Yes, change per the recommended solution.

Comment (bhatipradeep) [reply C++/competitive/LeetCode/coin_change_2.cpp:50]: okay

**Review by amanmehara — APPROVED** [review]
Thanks for contributing! 🎉

**Pull request merged**